src/astro/databases/base.py

```diff
@@ -0,0 +1,259 @@
+        metadata.create_all(self.sqlalchemy_engine, tables=[sqlalchemy_table])
+
+    def create_table_from_select_statement(
+        self,
+        statement: str,
+        target_table: Table,
+        parameters: Optional[dict] = None,
+    ) -> None:
+        """
+        Export the result rows of a query statement into another table.
+
+        :param statement: SQL query statement
+        :param target_table: Destination table where results will be recorded.
+        :param parameters: (Optional) parameters to be used to render the SQL query
+        """
+        statement = self._create_table_statement.format(
+            self.get_table_qualified_name(target_table), statement
+        )
+        self.hook.run(statement, parameters)
+
+    def drop_table(self, table: Table) -> None:
+        """
+        Delete a SQL table, if it exists.
+
+        :param table: The table to be deleted.
+        """
+        statement = self._drop_table_statement.format(
+            self.get_table_qualified_name(table)
+        )
+        self.hook.run(statement)
+
+    # ---------------------------------------------------------
+    # Table load methods
+    # ---------------------------------------------------------
+    def load_file_to_table(
+        self,
+        source_file: str,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:
+        """
+        Create a table with the file's contents.
+        If the table already exists, append or replace the content, depending on the value of `if_exists`.
+
+        :param source_file: Local or remote filepath
+        :param target_table: Table in which the file will be loaded
+        :param if_exists: Strategy to be used in case the target table already exists.
+        :param chunk_size: Specify the number of rows in each batch to be written at a time.
+        """
+        raise NotImplementedError
+
+    def load_pandas_dataframe_to_table(
+        self,
+        source_dataframe: pd.DataFrame,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+    ) -> None:
+        """
+        Create a table with the dataframe's contents.
+        If the table already exists, append or replace the content, depending on the value of `if_exists`.
+
+        :param source_dataframe: Local or remote filepath
+        :param target_table: Table in which the file will be loaded
+        :param if_exists: Strategy to be used in case the target table already exists.
+        :param chunk_size: Specify the number of rows in each batch to be written at a time.
+        """
+        source_dataframe.to_sql(
+            target_table.name,
+            con=self.sqlalchemy_engine,
+            if_exists=if_exists,
+            chunksize=chunk_size,
+            method="multi",
+            index=False,
+        )
+
+    def append_table(
```

Comment:
So far, I feel that any operation with a table/database argument should belong to `Database` - since they can be customized per `Database` if needed.

Most operators will build and offer end-user functionalities by using `Database` and any other abstractions we may need. Operators are still responsible for templating, handling users' arguments, instantiating tables if necessary, and sending info to xcom.

Do you have any concerns with this approach, @utkarsharma2 ?

Reply:
Not now, I think it would be cleaner to handle things per database if need be with this approach, compared to handling it in Operators.